dt-workflows/workflows-execution-handler.php

```diff
@@ -583,14 +583,87 @@ private static function action_append( $field_type, $field_id, $value ): array {
                 }
                 break;
             case 'comments':
+
+                // get post settings of the current post
+                $settings = DT_Posts::get_post_settings( $post['post_type'] );
+                // filter down to only reference fields - or can you just get connection_types?
+                $fields = array_filter( $settings['fields'], function( $obj ) {
+                    if ( $obj['type'] == 'connection' || $obj['type'] == 'user_select' ) {
+                        return true;
+                    }
+                    return false;
+                } );
+
+                // loop through each field, checking if the comment attempts to reference one of the fields
+                foreach ( $fields as $field => $data ) {
+                    if ( str_contains( $value, '{' . $field . '}' ) ) {
```

Comment:
I think we can do this in a little different way. Instead of looping over all of the fields, we could look for all of the tokens in the comment text and look for fields that match the token.

To do that, we can use RegEx (regular expressions). These are formulas you can write to parse any kind of text content. In this case, we can look for everything that's wrapped in curly brackets (`{...}`). I usually use https://regex101.com/ to write and test regex. In this case, the expression we can use is `\{(\S+?)\}`.

To use that in PHP, the sample code that that site generates is this:

```
$re = '/\{(\S+?)\}/m';
$str = 'Notify {subassigned} and {assigned_to}';

preg_match_all($re, $str, $matches, PREG_SET_ORDER, 0);

// Print the entire match result
var_dump($matches);
```

You can replace `$str` with the comment text. `$matches` will be an array where each item is a matched token in the comment. Each match will also be an array where the first item is the whole token with brackets, and the second item is just the text within the brackets. That second item should be the field name that you can use to check if there is a field on the post that matches, and then you can do the replacement.